systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptST.java

```diff
@@ -0,0 +1,143 @@
+        String[] backupCmd = new String[] {
+            PROJECT_HOME + "/tools/cold-backup/run.sh", "-by", "-n", NAMESPACE, "-c", CLUSTER_NAME, "-t", BACKUP_PATH
+        };
+        runExternalCommand(backupCmd);
+
+        // delete the namespace and restore
+        ResourceManager.deleteMethodResources();
+        ResourceManager.kubeClient().createNamespace(NAMESPACE);
+        LOGGER.info("Running restore procedure for {}/{}", NAMESPACE, CLUSTER_NAME);
+        String[] restoreCmd = new String[] {
+            PROJECT_HOME + "/tools/cold-backup/run.sh", "-ry", "-n", NAMESPACE, "-c", CLUSTER_NAME, "-s", BACKUP_PATH
+        };
+        runExternalCommand(restoreCmd);
+
+        // deploy the operator and wait for provisionig
+        installClusterOperator(NAMESPACE);
+        waitForKafkaReady(CLUSTER_NAME);
+        internalClientsPodName = deployInternalClients();
+        internalClients = buildInternalClients(internalClientsPodName, groupName, secondBatchSize);
+
+        // check consumer group offsets
+        LOGGER.info("Checking consumer group offsets after restore from pod: {}", internalClientsPodName);
+        Map<String, String> offsetsAfterRestore = internalClients.getCurrentOffsets();
+        logConsumerGroupOffsets(offsetsAfterRestore);
+        assertEquals(offsetsBeforeBackup, offsetsAfterRestore);
+
+        // check consumer group recovery
+        LOGGER.info("Checking consumer group recovery from pod: {}", internalClientsPodName);
+        assertEquals(secondBatchSize, internalClients.receiveMessagesPlain());
+
+        // check total number of messages
+        int batchSize = firstBatchSize + secondBatchSize;
+        internalClients = buildInternalClients(internalClientsPodName, newGroupName, batchSize);
+        LOGGER.info("Checking total number of messages from pod: {}", internalClientsPodName);
+        assertEquals(batchSize, internalClients.receiveMessagesPlain());
+    }
+
+    private String deployInternalClients() {
+        LOGGER.info("Deploying internal clients");
+        KafkaClientsResource.createAndWaitForReadiness(
+            KafkaClientsResource.deployKafkaClients(false, CLIENTS_NAME).build()
+        );
+        String internalClientsPodName = ResourceManager.kubeClient()
+            .listPodsByPrefixInName(CLIENTS_NAME).get(0).getMetadata().getName();
+        return internalClientsPodName;
+    }
+
+    private InternalKafkaClient buildInternalClients(String podName, String groupName, int messageCount) {
+        InternalKafkaClient clients = new InternalKafkaClient.Builder()
+            .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withTopicName(TOPIC_NAME)
+            .withUsingPodName(podName)
+            .withConsumerGroupName(groupName)
+            .withMessageCount(messageCount)
+            .build();
+        return clients;
+    }
```

Comment:
Well, you can't set topicName anf groupName on InternalKafkaClient, that's why I'm doing this.

Reply:
You can do
```
clients = clients.toBuilder()
            .withTopicName(TOPIC_NAME)
            .withConsumerGroupName(groupName)
            .build();
```
also, if you don't specify the `consumerGroup`, it will generate random one, so you don't need to bother with it.

Reply:
OK